force-app/main/default/classes/IndividualBucketAccount.cls

```diff
@@ -0,0 +1,63 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description The Individual Bucket Account Entity in NPSP.
+*/
+public inherited sharing class IndividualBucketAccount {
+
+    private Account record;
+
+    public static IndividualBucketAccount getInstance(){
+        List<Account> bucketAccounts = getBucketAccounts();
+        Account bucketAccount = (bucketAccounts.size() == 1) ? bucketAccounts.get(0) : null;
+        return new IndividualBucketAccount(bucketAccount);
+    }
+
+    private IndividualBucketAccount(Account anAccount){
+        this.record = anAccount;
+    }
+
+    public String name() {
+        return record.Name;
```

Comment:
If the `anAccount` parameter is null from the constructor after calling `getInstance()`, this could throw a null pointer. We should use `record?.Name` here.

Reply:
Good call, Daniel. After the next iteration of re-factor I think this won't be necessary because this entity should actually be created and inserted if it doesn't already exist when it is referenced. But for the time being there is a slight chance that this class could be used in the name property could be referenced when the account is null.  I added a test and the safe navigator. We can modify this test in the future in line with the re-factoring to use this class when creating new contacts and attaching them to the bucket account.